src/main/java/org/traccar/protocol/HuabaoProtocolDecoder.java

```diff
@@ -699,6 +699,17 @@ private Position decodeLocation(DeviceSession deviceSession, ByteBuf buf) {
                         }
                     }
                     break;
+                case 0xF4:
+                    while (buf.readerIndex() < endIndex) {
+                        WifiAccessPoint wifiAccessPoint = new WifiAccessPoint();
+                        wifiAccessPoint.setMacAddress(String.format("%02x:%02x:%02x:%02x:%02x:%02x",
+                                buf.readUnsignedByte(), buf.readUnsignedByte(), buf.readUnsignedByte(),
+                                buf.readUnsignedByte(), buf.readUnsignedByte(), buf.readUnsignedByte()));
+                        wifiAccessPoint.setSignalStrength((int) buf.readByte());
+                        network.addWifiAccessPoint(wifiAccessPoint);
+                    }
+                    position.setNetwork(network);
```

Comment:
I think instead of calling `setNetwork` in multiple places, it should also be moved to the top level. Something similar to what we do in other places, link [this](https://github.com/traccar/traccar/blob/master/src/main/java/org/traccar/protocol/EelinkProtocolDecoder.java#L241-L243).